frontend/src/components/header.js

```diff
@@ -42,65 +44,77 @@ BootstrapNavLink.propTypes = {
  * @param {object[]} props.routes
  * @returns
  */
+
 export function Header(props) {
     const { routes = [], infoComponents = null } = props;
+    let match = useRouteMatch("/:mainRoute/:subRoute?");
+    const { mainRoute } = match.params;
 
     if (routes.length === 0) {
         return <div>No Routes in Header</div>;
     }
 
+    const activeMainRoutes = routes.map((route) => (
+        <BootstrapNavItem
+            eventKey={route.route}
+            to={route.route}
+            key={route.route}
+            className="primary"
+        >
+            <h5>{route.name}</h5>
+        </BootstrapNavItem>
+    ));
+
+    const activeSubRoutes = routes
+        .filter((route) => route.route.substring(1) === mainRoute)
+        .map((route) =>
+            (route.subroutes || []).map((subroute) => {
+                const fullroute = `${route.route}${subroute.route}`;
+                console.log(route, subroute);
+
+                return (
+                    <BootstrapNavItem
+                        to={fullroute}
+                        title={subroute.name}
+                        key={fullroute}
+                        className="secondary"
+                    >
+                        <h5>
+                            <small>{subroute.name}</small>
+                        </h5>
+                    </BootstrapNavItem>
+                );
+            })
+        );
+
     return (
-        <Navbar bg="light" variant="light">
-            <Navbar.Brand>
-                <NavDropdown
-                    title={routes.map((route) => (
-                        <Route path={route.route} key={route.route}>
-                            {route.name}
-                        </Route>
-                    ))}
-                >
-                    {routes
-                        .filter((route) => !route.hidden)
-                        .map((route) => (
-                            <NavDropdown.Item
-                                key={route.route}
-                                as="span"
-                                tabIndex="0"
-                            >
-                                <BootstrapNavLink
-                                    to={route.route}
-                                    title={route.description}
-                                >
-                                    {route.name}
-                                </BootstrapNavLink>
-                            </NavDropdown.Item>
+        <Tab.Container activeKey={mainRoute} defaultActiveKey={"tapp"}>
+            <Row className="justify-content-between pr-3 ">
+                <Col md={"auto"}>
+                    <Nav
+                        className="flex-row pt-3 pr-3 pb-0 pl-2"
```

Comment:
Create a new class name and move this to CSS. Only use the inline-bootstrap styles for minor things (e.g., you only need one or two bootstrap classes)